vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.html

```diff
@@ -24,309 +27,339 @@
+                		makeSpaceLinksEditable(label, link.link.id, rotation ,targetSpace,link.positionX,link.positionY,link.id,type);
+                	});
+                }
+            })
+            moduleLinks.forEach(function(link,index){
+                if(link != null){
+                	var modalModuleLink;
+                	if(link.type=="ALERT"){
+                	    modalModuleLink = $('<div class="alert alert-primary moduleLink-'+link.link.id+'" role="alert" data-link-id="'+link.link.id+'"><p class="label-'+link.link.id+'">'+link.link.name+'</p></div>');
+                	}
+                	else if(link.type=="IMAGE" && link.image != null){
+                		var modalModuleImageURL = [[@{/api/image/}]]+link.image.id;
+                		modalModuleLink = $('<img id="'+link.image.id+'" data-link-id="'+link.link.id+'" class="moduleLink-'+link.link.id+'" src="'+modalModuleImageURL+'" />');
+                	}
+                	else{
+                	    modalModuleLink = $('<span data-link-id="'+link.link.id+'" class="moduleLink-'+link.link.id+' Info_cz_Class"><svg class="Ellipse_8_c"><ellipse fill="rgba(222,222,222,1)" class="Ellipse_8_c_Class" rx="14.5" ry="14.5" cx="14.5" cy="14.5"></ellipse></svg><svg class="Ellipse_10_c"><ellipse fill="rgba(240,240,240,1)" class="Ellipse_10_c_Class" rx="12.5" ry="12.5" cx="12.5" cy="12.5"></ellipse></svg><svg class="Ellipse_9_c"><ellipse fill="rgba(255,255,255,1)" class="Ellipse_9_c_Class" rx="10.5" ry="10.5" cx="10.5" cy="10.5"></ellipse></svg><i class="fas fa-info fa-lg Icon_awesome_info_staff_m"></i><p class="label-'+link.link.id+'" data-link-id="'+link.link.id+'"><span class="tooltiptext">'+link.link.name+'</span></p></span>');
+                	}
+                	setLinkPosition(modalModuleLink,link,false,true,false);
+                	var inputLinkedModule = document.createElement("input");
+                	inputLinkedModule.setAttribute("type", "hidden");
+                	inputLinkedModule.setAttribute("id", "moduleLinkTarget-"+link.link.id);
+                	inputLinkedModule.setAttribute("value", link.link.module.id);
+                	$("#space").append(inputLinkedModule);
+                	$("#space").append(modalModuleLink);
+                	$('[data-link-id="'+link.link.id+'"]').css('cursor', 'pointer');
+                	$('[data-link-id="'+link.link.id+'"]').click(function(e) {
+                	    var rotation = document.getElementById("linkRotation-"+link.link.id).value;
+                	    var label = document.getElementById("linkLabel-"+link.link.id).value;
+                	    var type = document.getElementById("linkType-"+link.link.id).value;
+                	    var targetModule = document.getElementById("moduleLinkTarget-"+link.link.id).value;
+                		makeModuleLinksEditable(label, link.link.id,rotation,targetModule,link.positionX,link.positionY,link.id, type);
+                	});

```

Comment:
this all looks like a lot of repeated code. You can probably extract some methods and call them with the write parameters instead of repeating almost the same code three times.